src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ILGeneratorImpl.cs

```diff
@@ -817,4 +855,40 @@ internal LabelInfo(LabelHandle metaLabel)
         internal int _startDepth; // Stack labelStartDepth, with -1 meaning unknown.
         internal LabelHandle _metaLabel;
     }
+
+    internal sealed class Scope
+    {
+        public Scope(int offset, Scope? parent)
+        {
+            _startOffset = offset;
+            _parent = parent;
+        }
+
+        internal Scope? _parent;
+        internal List<Scope>? _children;
+        internal List<LocalBuilder>? _locals;
+        internal List<string>? _importNamespaces;
+        internal int _startOffset;
+        internal int _endOffset;
```

Comment:
Are we preferring fields as opposed to properties? If we are going with non-private fields I think I'd prefer them to not start with an `_` prefix. Unsure if this is a common pattern or not.

Reply:
Using properties would allow us to set the property to non `null` and hide that away instead of it leaking to other places.

Reply:
> I think I'd prefer them to not start with an _ prefix.

The libraries [coding style](https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md) is to use `_` prefix for both private and internal fields.

Reply:
> Are we preferring fields as opposed to properties

Have no strong preference, used the same pattern already or other internal types, can change to property if needed.

> Using properties would allow us to set the property to non null and hide that away instead of it leaking to other places.

Not sure I understood, but the null value is used for deciding if they should be emitted

Reply:
Okay. This is slightly more nuanced than I thought. Let's leave it.